packages/sdk/src/swap/assets.ts

```diff
@@ -1,62 +1,48 @@
 import { getTokenContractAddress } from '@/shared/contracts';
 import {
-  Asset,
-  AssetAndChain,
-  assetChains,
-  assetDecimals,
-  Assets,
+  InternalAsset,
   ChainflipNetwork,
   isTestnet,
+  readChainAssetMap,
+  assetConstants,
 } from '@/shared/enums';
 import type { Environment } from '@/shared/rpc';
-import { readAssetValue } from '@/shared/rpc/utils';
 import type { AssetData } from './types';
 
 type AssetFn = (
   network: ChainflipNetwork,
   env: Pick<Environment, 'swapping' | 'ingressEgress'>,
 ) => AssetData;
 
-const assetName: Record<Asset, string> = {
-  [Assets.ETH]: 'Ether',
-  [Assets.USDC]: 'USDC',
-  [Assets.FLIP]: 'FLIP',
-  [Assets.DOT]: 'Polkadot',
-  [Assets.BTC]: 'Bitcoin',
-};
-
 const assetFactory =
-  (asset: Asset): AssetFn =>
-  (network, env) => {
-    const assetAndChain = { asset, chain: assetChains[asset] } as AssetAndChain;
-
-    return {
-      id: asset,
```

Comment:
removing the id here is a small bc break, but leaving the id like it is (`ETH`, `USDC`, ...) would be confusing bc it will not identify the asset anymore (there will be two `USDC` assets with arbitrum)

Reply:
Extracted into https://github.com/chainflip-io/chainflip-sdk-monorepo/pull/557